akvo/templates/project_directory.html

```diff
@@ -264,6 +271,26 @@ <h4 class="detailedInfo">{% trans "Finance" %}</h4>
         }
     </script>
 
+    {# More partners data #}
+    {% for project in page %}
+    <script type="application/json" id="more-partners-{{project.id}}">
+    [
+    {% for partnership in project.partnerships.distinct %}
```

Comment:
.distinct seems to destroy the prefetch here and I can't see that it's needed. I replaced it with .all and got an order of magnitude faster response. 

If you think it should be distinct it has to be part of the query using the [Prefetch](https://docs.djangoproject.com/en/1.7/ref/models/queries/#django.db.models.Prefetch) class I think.